metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -773,10 +773,18 @@ static void validateNewTopicNames(Map<String, ApiError> topicErrors,
         for (CreatableTopic topic : topics) {
             if (topicErrors.containsKey(topic.name())) continue;
             Map<String, Entry<OpType, String>> topicConfigs = new HashMap<>();
+            List<String> nullConfigs = new ArrayList<>();
             for (CreateTopicsRequestData.CreateableTopicConfig config : topic.configs()) {
-                topicConfigs.put(config.name(), new SimpleImmutableEntry<>(SET, config.value()));
+                if (config.value() == null) {
+                    nullConfigs.add(config.name());
+                } else {
+                    topicConfigs.put(config.name(), new SimpleImmutableEntry<>(SET, config.value()));
+                }
             }
-            if (!topicConfigs.isEmpty()) {
+            if (!nullConfigs.isEmpty()) {
+                topicErrors.put(topic.name(), new ApiError(Errors.INVALID_REQUEST,
```

Comment:
On a related note, I opened https://github.com/apache/kafka/pull/12162. I wonder if we should call this INVALID_CONFIG instead?

Reply:
`INVALID_CONFIG` is more suitable here, I also changed the exception in `ControllerConfigurationValidator`.